client/engine/engine.go

```diff
@@ -367,7 +367,8 @@ func (e *Engine) handleObjectiveRequest(or protocols.ObjectiveRequest) (EngineEv
 			return EngineEvent{}, fmt.Errorf("handleAPIEvent: Could not create objective for %+v: %w", request, err)
 		}
 		// Only Alice or Bob care about registering the objective and keeping track of vouchers
-		if vfo.MyRole == payments.PAYEE_INDEX || vfo.MyRole == payments.PAYER_INDEX {
+		lastParticipant := uint(len(vfo.V.Participants) - 1)
+		if vfo.MyRole == lastParticipant || vfo.MyRole == payments.PAYER_INDEX {
```

Comment:
Isn't it the case that only Alice initiates the VF? If so, should this be replaced with a `if myRole != payer return error` sort of check?

Reply:
> Isn't it the case that only Alice initiates the VF?

I think that's true at the moment due to our API. `CreateVirtualPaymentChannel` assumes the caller is `alice`.  It might be nice to allow the engine to handle objective requests from other participants in case our API changes in the future?

Reply:
Will leave it alone for now.